src/lib/commands.js

```diff
@@ -0,0 +1,271 @@
+
+const COOKIE_VERSION = 2;
+const CMP_ID = 280;
+const CMP_VERSION = 2;
+
+const createCommands = (store) => {
+	let vendorList;
+
+	const getVendorList = (callback) => {
+		if (vendorList) {
+			callback(vendorList);
+		} else if (store.vendorList && Object.keys(store.vendorList).length) {
+			vendorList = JSON.parse(JSON.stringify(store.vendorList));
+			callback(vendorList);
+		} else {
+			const { getVendorList } = config;
+
+			if (getVendorList) {
+				getVendorList((err, data) => {
+					if (err) {
+						callback({}, err);
+					} else {
+						vendorList = data;
+						callback(vendorList);
+					}
+				});
+			}
+		}
+	};
+
+	const vectorToObject = (set) => {
+		let obj = {};
+
+		set.forEach((hasConsent, id) => {
+			obj[id] = hasConsent;
+		});
+
+		return obj;
+	};
+
+	 /**
+	 * Build consent fields object v1 from data that has already been persisted.
+	 */
+	const getConsentFieldsObject = () => {
+		const {
+			persistedConsentData
+		} = store;
+
+		const {
+			created = new Date(),
+			lastUpdated = new Date(),
+			cmpId_: cmpId = store.tcModel.cmpId,
+			cmpVersion_: cmpVersion = 1,
+			consentScreen_: consentScreen = store.tcModel.consentScreen,
+			consentLanguage_: consentLanguage = store.tcModel.consentLanguage,
+			vendorListVersion_: vendorListVersion,
+		} = persistedConsentData;
+
+		return {
+			cmpId,
+			cmpVersion,
+			consentLanguage,
+			consentScreen,
+			created,
+			vendorListVersion,
+			lastUpdated,
+		};
+	};
+
+	const generateConsentStringV1 = (data = {}, callback) => {
+
+		getVendorList((vendorList, err) => {
+			if (err) {
+				callback({}, false);
+			}
+			if (vendorList) {
+				const {
+					persistedConsentData,
+				} = store;
+
+				const {
+					created,
+					lastUpdated,
+					consentLanguage_: consentLanguage,
+					consentScreen_: consentScreen,
+					version_: version = 1,
+					cmpId_: cmpId = 280
+				} = persistedConsentData;
+
+				const {
+					selectedVendorIds = [],
+					selectedPurposeIds = [],
+					maxVendorId,
+					cookieVersion,
+					vendorListVersion,
+					cmpVersion
+				} = data;
+
+				const consentData = {
+					cmpId,
+					maxVendorId,
+					cookieVersion,
+					cmpVersion,
+					vendorListVersion,
+					created,
+					lastUpdated,
+					consentLanguage,
+					consentScreen,
+					version,
+					selectedVendorIds: new Set(arrayFrom(selectedVendorIds)),
+					selectedPurposeIds: new Set(arrayFrom(selectedPurposeIds))
+				};
+
+				const valid = (data = {}) => [
+					'cmpId', 'cmpVersion', 'consentLanguage', 'consentScreen', 'cookieVersion', 'created', 'lastUpdated',
+					'maxVendorId', 'selectedPurposeIds', 'selectedVendorIds', 'vendorListVersion'
+				].every(prop => data.hasOwnProperty(prop));
+
+				callback(valid(consentData) ? encodeVendorConsentData(consentData) : undefined);
+			}});
+	};
+
+	const getConsentFieldsV1 = (callback, params) => {
+		const data = getConsentFieldsObject();
+		const now = new Date();
```

Comment:
This function should return `undefined` as created and lastUpdated when persistedData doesn't exist.